pkg/accumulator/resaccumulator_test.go

```diff
@@ -179,6 +179,141 @@ func TestResolveVarsVarNeedsDisambiguation(t *testing.T) {
 	}
 }
 
+func TestResolveVarsAllowableConflict(t *testing.T) {
```

Comment:
please make some little private factory methods so the test reads more like:

```
rmFooApple := makeResourceMap("envFoo", "dataValApple")
rmBarApple := makeResourceMap("envBar", "dataValApple")
rmBarPeach := makeResourceMap("envBar", "dataValPeach")

vFoo := makeVar("envFoo")
vBar := makeVar("envBar")

ac0 := new accumulator
ac0.Append(rmFooApple)
ac0.MergeVar(vFoo)

ac1 := new accumulator
ac1.Append(rmBarApple)
ac1.MergeVar(vBar)

assert no error merging ac1 into ac0 because values the same

ac2 := new accumulator
ac1.Append(rmBarPeach)
ac1.MergeVar(vBar)
// works


assert error merging ac2 into ac1 because values different
assert error merging ac2 into ac0 because values different
```


Also please - for clarity - write this test as a distinct PR, where presumably all accumulation mergers fail,

we'll merge that test, 

then rebase this PR to include that test, and then this PR adds the new code and a minor tweak to the test to show what doesn't error anymore.

Reply:
Thanks for the clear direction. I'll get this done before EOD tomorrow.